sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsBufferCatalog.scala

```diff
@@ -864,30 +865,15 @@ object RapidsBufferCatalog extends Logging {
   }
 
   private def closeImpl(): Unit = synchronized {
-    if (_singleton != null) {
-      _singleton.close()
-      _singleton = null
-    }
-
-    if (memoryEventHandler != null) {
-      // Workaround for shutdown ordering problems where device buffers allocated with this handler
-      // are being freed after the handler is destroyed
-      //Rmm.clearEventHandler()
-      memoryEventHandler = null
-    }
+    val memoryEventHandlerCloser: AutoCloseable = () =>
+      if (memoryEventHandler != null) {
+        // Workaround for shutdown ordering problems where device buffers allocated
+        // with this handler are being freed after the handler is destroyed
+        //Rmm.clearEventHandler()
+        memoryEventHandler = null
+      }
```

Comment:
nit: does not safe lines of code and creates an additional class for lambda, maybe just leave it as is?

Reply:
Same here. These resources should be closed even if one of them fails to close. The use of lambda was added to keep the original order of clearing resources.